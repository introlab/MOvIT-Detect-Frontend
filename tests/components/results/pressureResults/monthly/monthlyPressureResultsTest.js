/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import MonthlyPressureResults from '../../../../../src/components/results/pressureResults/monthly/monthlyPressureResults';


Enzyme.configure({ adapter: new Adapter() });

const month = '1';

describe('MonthlyPressureResults Tests', () => {
  let wrapper;

  const initialState = {
    applicationReducer: { language: 'FR' },
    recommendationReducer: {
      reduceWeight: true,
      reduceSlidingRest: true,
      reduceSlidingMoving: true,
    },
  };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    month,
  };

  beforeEach(() => {
    wrapper = shallow(<MonthlyPressureResults store={store} {...props} />).dive();
  });

  it('should have proptypes', () => {
    const actualValue = MonthlyPressureResults.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      month: PropTypes.string.isRequired,
      reduceWeight: PropTypes.bool,
      reduceSlidingMoving: PropTypes.bool,
      reduceSlidingRest: PropTypes.bool,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    wrapper.setState({
      isLoaded: true,
      hasErrors: false,
    });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
