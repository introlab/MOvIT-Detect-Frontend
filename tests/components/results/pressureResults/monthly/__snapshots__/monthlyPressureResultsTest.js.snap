// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyPressureResults Tests should match the snapshot 1`] = `
<div>
  <div
    className="col-lg-2 leftMenu"
  >
    <ul
      className="graphlist"
    >
      <li
        className="graphLink"
      >
        <a
          href="results/pressure#monthlySitting"
        >
          results.graphicsLink.sittingTime.FR
        </a>
      </li>
      <li
        className="graphLink"
      >
        <a
          href="results/pressure#reduceWeight"
        >
          dailyResults.pressure.FR
        </a>
      </li>
    </ul>
  </div>
  <div
    className=" col-lg-10 offset-lg-3 results resultsContainer"
  >
    <div
      className="col-lg-8 graphic"
    >
      <div>
        <Connect(MonthlySittingTime)
          month="1"
        />
        <div
          id="reduceWeight"
        >
          <RecGoalChart
            condition={true}
            goalData={
              Object {
                "datasets": Array [
                  Object {
                    "borderColor": "red",
                    "data": Array [
                      36,
                      40,
                      27,
                      38,
                      42,
                      55,
                      40,
                      28,
                      32,
                      26,
                      25,
                      28,
                      31,
                      22,
                      25,
                      34,
                      36,
                      30,
                      21,
                      24,
                      26,
                      28,
                      31,
                      32,
                      8,
                      0,
                      26,
                      30,
                      21,
                      24,
                    ],
                    "fill": true,
                    "label": "monthlyResults.pressure.tiltMade.FR",
                    "lineTension": 0,
                  },
                  Object {
                    "borderColor": "blue",
                    "data": Array [
                      26,
                      28,
                      31,
                      32,
                      8,
                      34,
                      36,
                      30,
                      21,
                      24,
                      55,
                      40,
                      28,
                      32,
                      26,
                      25,
                      28,
                      31,
                      22,
                      25,
                      0,
                      26,
                      30,
                      21,
                      24,
                      36,
                      40,
                      27,
                      38,
                      42,
                    ],
                    "fill": false,
                    "label": "monthlyResults.pressure.tiltGoal.FR",
                    "lineTension": 0,
                  },
                ],
                "labels": Array [
                  "1",
                  "2",
                  "3",
                  "4",
                  "5",
                  "6",
                  "7",
                  "8",
                  "9",
                  "10",
                  "11",
                  "12",
                  "13",
                  "14",
                  "15",
                  "16",
                  "17",
                  "18",
                  "19",
                  "20",
                  "21",
                  "22",
                  "23",
                  "24",
                  "25",
                  "26",
                  "27",
                  "28",
                  "29",
                  "30",
                ],
              }
            }
            goalTitle="monthlyResults.pressure.personal.FR"
            recData={
              Object {
                "datasets": Array [
                  Object {
                    "borderColor": "red",
                    "data": Array [
                      36,
                      40,
                      27,
                      38,
                      42,
                      55,
                      40,
                      28,
                      32,
                      26,
                      25,
                      28,
                      31,
                      22,
                      25,
                      34,
                      36,
                      30,
                      21,
                      24,
                      26,
                      28,
                      31,
                      32,
                      8,
                      0,
                      26,
                      30,
                      21,
                      24,
                    ],
                    "fill": true,
                    "label": "monthlyResults.pressure.tiltMade.FR",
                    "lineTension": 0,
                  },
                  Object {
                    "borderColor": "blue",
                    "data": Array [
                      26,
                      28,
                      31,
                      32,
                      8,
                      34,
                      36,
                      30,
                      21,
                      24,
                      55,
                      40,
                      28,
                      32,
                      26,
                      25,
                      28,
                      31,
                      22,
                      25,
                      0,
                      26,
                      30,
                      21,
                      24,
                      36,
                      40,
                      27,
                      38,
                      42,
                    ],
                    "fill": false,
                    "label": "monthlyResults.pressure.tiltGoal.FR",
                    "lineTension": 0,
                  },
                ],
                "labels": Array [
                  "1",
                  "2",
                  "3",
                  "4",
                  "5",
                  "6",
                  "7",
                  "8",
                  "9",
                  "10",
                  "11",
                  "12",
                  "13",
                  "14",
                  "15",
                  "16",
                  "17",
                  "18",
                  "19",
                  "20",
                  "21",
                  "22",
                  "23",
                  "24",
                  "25",
                  "26",
                  "27",
                  "28",
                  "29",
                  "30",
                ],
              }
            }
            recTitle="monthlyResults.pressure.recommended.FR"
            title="monthlyResults.pressure.FR"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
